sigmastate/src/main/scala/sigmastate/Values.scala

```diff
@@ -1175,13 +1170,7 @@ object Values {
 
     protected final override def eval(env: DataEnv)(implicit E: ErgoTreeEvaluator): Any = {
       addCost(FuncValue.costKind)
-      if (args.length == 0) {
```

Comment:
It is not breaking change, because it is a change in the new interpreter. This code shouldn't be there in a first place, so consider it a cleanup. And there are tests that cover this.
Also v4.x only support single argument lambdas.

Reply:
Serializer allows for args.length != 1, so where args.length != 1 prevented before this code being executed?

Reply:
In v4.x each FuncValue should be processed by this case https://github.com/ScorexFoundation/sigmastate-interpreter/blob/e0905cc8c135fc4a6669f04636dd888aaa11572b/sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala#L1760
And each Apply node by this  https://github.com/ScorexFoundation/sigmastate-interpreter/blob/e0905cc8c135fc4a6669f04636dd888aaa11572b/sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala#L1403

They only support single argument, all others will go to https://github.com/ScorexFoundation/sigmastate-interpreter/blob/e0905cc8c135fc4a6669f04636dd888aaa11572b/sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala#L1857

So, even if ErgoTree can be deserialized, it still can be rejected.